### PR TITLE
Proxyless: support file protocol, 'disablePageReloads' option

### PR DESCRIPTION
**Pull request by miherlosev**
[Closes DevExpress/testcafe-private#19]

Changes:
* support file protocol in `proxyless` mode.
* support the `disablePageReloads` option in `proxyless` mode.
* move service routes to the separate file and use in necessary places.
* refactoring

### CODE REVIEW

File: src/proxyless/resource-injector.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import debug from 'debug';
 import { StatusCodes } from 'http-status-codes';
```
Comment (Aleksey28) [inline src/proxyless/resource-injector.ts:20]: Why didn't you move calls of other `client` methods to `utils`?

File: src/client/driver/driver.js [hunk 2]
```diff
@@ -118,6 +118,7 @@ import createErrorCtorCallback, {
 import './command-executors/action-executor/actions-initializer';
 import { shouldSkipJsError } from './process-skip-js-errors';
 
+
```
Comment (Aleksey28) [inline src/client/driver/driver.js:121]: Removed please

File: src/client/driver/driver.js [hunk 3]
```diff
@@ -1490,14 +1478,13 @@ export default class Driver extends serviceUtils.EventEmitter {
...
 
         return browser
-            .checkStatus(urls, hammerhead.createNativeXHR, { manualRedirect: true, proxyless: this.proxyless })
-            .then(({ command, redirecting }) => {
-                const isSessionChange = redirecting && command.url.indexOf(this.testRunId) < 0;
+            .checkStatus(urls, hammerhead.createNativeXHR, { manualRedirect: true, proxyless: this.options.proxyless })
+            .then(({ command }) => {
+                const isSessionChange = command.testRunId !== this.testRunId;
 
                 if (isSessionChange) {
```
Comment (Aleksey28) [inline src/client/driver/driver.js:1489]: I think it needs to merge this `if...else` block with the next one

File: test/functional/fixtures/api/es-next/take-screenshot/test.js [hunk 4]
```diff
@@ -50,6 +50,8 @@ const getReporter = function (scope) {
     });
 };
 
+const itFn = config.proxyless ? it.skip : it;
```
Comment (Aleksey28) [inline test/functional/fixtures/api/es-next/take-screenshot/test.js:53]: The project has this condition in other tests. Can you move it to utils and replace `if (!config.proxyless)` everywhere?

**Review by Aleksey28 — CHANGES_REQUESTED** [review]

File: src/client/driver/driver.js (hunk 2 above)
Comment (miherlosev) [reply src/client/driver/driver.js:121]: We usually add 2 new lines after imports.
So, this new line here is valid.

File: src/proxyless/resource-injector.ts (hunk 1 above)
Comment (miherlosev) [reply src/proxyless/resource-injector.ts:20]: The `redirect` and `navigateTo` methods are common utility methods. They are used in a few places.
Note: the `redirect` method is used in one place but it is analog of `navigateTo` method.
If you know other utility methods, please write them.

**Review by Aleksey28 — APPROVED** [review]

**Pull request merged**